symbology/create-symbols-from-web-styles/src/main/java/com/esri/samples/create_symbols_from_web_styles/CreateSymbolsFromWebStylesSample.java

```diff
@@ -0,0 +1,316 @@
+
+  /**
+   * Creates a legend on the UI for the symbol styles.
+   *
+   */
+  private void setupLegend() {
+
+    // create a grid pane and set the size, background color and spacing
+    gridPane = new GridPane();
+    gridPane.getColumnConstraints().addAll(Arrays.asList(new ColumnConstraints(70), new ColumnConstraints(120)));
+    gridPane.setMaxWidth(175);
+    gridPane.setMaxHeight(570);
+    gridPane.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(255,255,255, 0.9)"), CornerRadii.EMPTY,
+      Insets.EMPTY)));
+    gridPane.setPadding(new Insets(10));
+    gridPane.setVgap(12);
+
+    // create a label to display the symbol style name as the title of the legend and add to the grid pane
+    Label legendTitle = new Label("Style: " + symbolStyle.getStyleName());
+    legendTitle.setStyle("-fx-font-weight: bold");
+    gridPane.add(legendTitle, 0, 0,2,1);
+
+    // create labels for the column headings and add to the grid pane
+    Label symbolTitle = new Label("Symbol");
+    symbolTitle.setStyle("-fx-font-weight: bold");
+    Label nameTitle = new Label("Name");
+    nameTitle.setStyle("-fx-font-weight: bold");
+    gridPane.add(symbolTitle, 0, 1);
+    gridPane.add(nameTitle, 1, 1);
+  }
+
+  /**
+   * Returns a list of categories to be matched to a symbol name.
+   *
+   * @param symbolName the name of a symbol from a symbol style
+   * @return categories a list of categories matched to the provided symbol name
+   */
+  private List<String> mapSymbolNameToField(String symbolName) {
+
+    List<String> categories = new ArrayList<>();
+
+    switch (symbolName) {
+      case "atm":
+        categories.add("Banking and Finance");
+        break;
+      case "beach":
+        categories.add("Beaches and Marinas");
+        break;
+      case "campground":
+        categories.add("Campgrounds");
+        break;
+      case "city-hall":
+        categories.addAll(Arrays.asList("City Halls", "Government Offices"));
+        break;
+      case "hospital":
+        categories.addAll(Arrays.asList("Hospitals and Medical Centers", "Health Screening and Testing", "Health Centers",
+          "Mental Health Centers"));
+        break;
+      case "library":
+        categories.add("Libraries");
+        break;
+      case "park":
+        categories.add("Parks and Gardens");
+        break;
+      case "place-of-worship":
+        categories.add("Churches");
+        break;
+      case "police-station":
+        categories.add("Sheriff and Police Stations");
+        break;
+      case "post-office":
+        categories.addAll(Arrays.asList("DHL Locations", "Federal Express Locations"));
+        break;
+      case "school":
+        categories.addAll(Arrays.asList("Public High Schools", "Public Elementary Schools", "Private and Charter Schools"));
+        break;
+      case "trail":
+        categories.add("Trails");
+        break;
+    }
+    return categories;
+  }
+
+  /**
+   * Returns an imageView populated with a symbol.
```

Comment:
We should probably either use plain-english, or the full class names for these comments? So either
```suggestion
   * Returns an ImageView populated with a Symbol.
```
or:
```suggestion
   * Returns an image view populated with a symbol.
```